### PR TITLE
Task 50883 : Retrieve rewards count from user's profile

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**
I added an API to return the sum of rewards for each user, and I used EntityTag to improve the performance

### CODE REVIEW

File: wallet-webapps/src/main/webapp/vue-app/wallet-overview/components/WalletOverview.vue [hunk 1]
```diff
@@ -120,6 +123,25 @@ export default {
...
+        if (resp && resp.ok) {
+          return resp.json();
+        } else {
+          throw new Error('Error count rewards');
+        }
+      }).catch((error) => {
+        console.error(error);
+        this.error = this.$t('exoplatform.wallet.error.errorCountRewards');
+      });
+    },
```
Comment (ahamdi) [inline wallet-webapps/src/main/webapp/vue-app/wallet-overview/components/WalletOverview.vue]: Could you export this function in a dedicated JS API file for rewarding

File: wallet-api/src/main/java/org/exoplatform/wallet/reward/rest/RewardReportREST.java [hunk 2]
```diff
@@ -127,4 +127,31 @@ public Response listRewards(@ApiParam(value = "limit of items to load", required
...
+  @ApiOperation(value = "Return sum of rewards for current user", httpMethod = "GET", produces = "application/json", response = Response.class, notes = "return sum of rewards per user")
+  @ApiResponses(value = {
+          @ApiResponse(code = HTTPStatus.OK, message = "Request fulfilled"),
+          @ApiResponse(code = HTTPStatus.UNAUTHORIZED, message = "Unauthorized operation"),
+          @ApiResponse(code = 500, message = "Internal server error") })
+  public Response countRewards() {
+    try {
+      Double sumRewards = rewardReportService.countRewards(WalletUtils.getCurrentUserId());
+      JSONObject result = new JSONObject();
+      result.put("sumRewards", sumRewards);
```
Comment (ahamdi) [inline wallet-api/src/main/java/org/exoplatform/wallet/reward/rest/RewardReportREST.java]: Could you add Etag to this request to cache it on browser side 
an example https://github.com/meeds-io/social/blob/develop/component/service/src/main/java/org/exoplatform/social/rest/impl/identity/IdentityRestResourcesV1.java#L134

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: wallet-webapps/src/main/webapp/vue-app/wallet-overview/components/WalletOverview.vue [hunk 3]
```diff
@@ -75,6 +74,7 @@ export default {
     currencyName: null,
     currencySymbol: null,
     rewardBalance: 0,
```
Comment (ahamdi) [inline wallet-webapps/src/main/webapp/vue-app/wallet-overview/components/WalletOverview.vue]: Do we still have the rewardbalance property ? if not let's remove it.

File: wallet-webapps/src/main/webapp/vue-app/wallet-overview/components/WalletOverview.vue (hunk 1 above)
Comment (Rdidmohamedelhadi) [reply wallet-webapps/src/main/webapp/vue-app/wallet-overview/components/WalletOverview.vue]: done !

**Review by ahamdi — APPROVED** [review]

**Pull request merged**